exercises/dominoes/dominoes_test.rb

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env ruby
+gem 'minitest', '>= 5.0.0'
+require 'minitest/autorun'
+require_relative 'dominoes'
+
+# Test data version:
+# 018c48d
+class DominoesTest < Minitest::Test
+  def assert_correct_chain(input_dominoes, output_chain)
+    refute_nil output_chain, "There should be a chain for #{input_dominoes}"
+
+    input_normal = input_dominoes.map(&:sort).sort
+    output_normal = output_chain.map(&:sort).sort
+    assert_equal input_normal, output_normal,
+      'Dominoes used in the output must be the same as the ones given in the input'
+
+    output_chain.each_cons(2).with_index { |(d1, d2), i|
+      _, d1_right = d1
+      d2_left, _ = d2
+      assert_equal d1_right, d2_left,
+        "In chain #{output_chain}, right end of domino #{i} (#{d1}) and left end of domino #{i + 1} (#{d2}) must match"
+    }
+
+    return if output_chain.empty?
```

Comment:
Move this up above line 17. If we're short-circuiting short circuit as early as possible.